// @shopify/polaris/esnext/components/Portal/Portal.js
import React from 'react';
import { createPortal } from 'react-dom';
import { createUniqueIDFactory } from '@shopify/javascript-utilities/other';
import { themeProvider } from '../shared';
const getUniqueID = createUniqueIDFactory('portal-');
export class Portal extends React.PureComponent {
    constructor() {
        super(...arguments);
        this.state = { isMounted: false };
        this.portalId = this.props.idPrefix !== ''
            ? `${this.props.idPrefix}-${getUniqueID()}`
            : getUniqueID();
    }
    componentDidMount() {
        this.portalNode = document.createElement('div');
        this.portalNode.setAttribute('data-portal-id', this.portalId);
        this.portalContainerNode = document.querySelector(`${themeProvider.selector}`);
        if (this.portalContainerNode != null) {
            this.portalContainerNode.appendChild(this.portalNode);
        }
        this.setState({ isMounted: true });
    }
    componentDidUpdate(_, prevState) {
        const { onPortalCreated = noop } = this.props;
        if (!prevState.isMounted && this.state.isMounted) {
            onPortalCreated();
        }
    }
    componentWillUnmount() {
        if (this.portalContainerNode != null) {
            this.portalContainerNode.removeChild(this.portalNode);
        }
    }
    render() {
        return this.state.isMounted
            ? createPortal(this.props.children, this.portalNode)
            : null;
    }
}
Portal.defaultProps = { idPrefix: '' };
function noop() { }
